### PR TITLE
FFFra/account card component

### LINKED ISSUE #16028: Implement quo2 - Wallet - account card component
Implement the wallet accounts slider as per designs:

https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-UI-for-Mobile?type=design&node-id=37-142126&t=VMPvPP4JUgrCdzAe-0

Please place the implementation in the quo2 preview.
Have a word/chat with our designers before starting to implement to clarify the interactions.

You should be able to swipe the list horizontally.

Please componentize the various parts into atomic components.

### PR DESCRIPTION
**Pull request by FFFra**
fixes #16028

### Summary

This PR enhances the wallet accounts functionality, featuring a new slider, add-account feature with alert notifications, and dynamic preview mode. It supports both light and dark modes, includes comprehensive unit tests, and aligns with the provided design specifications. Please refer to the attached demo for a practical demonstration. Feedback and suggestions are most welcome.

...

### Testing notes
All tests are expected to pass successfully

#### Platforms
- Android
- iOS

##### Functional
- wallet / transactions

### Steps to Test
Please navigate to "Profile and Quo.2.0 Preview" and locate the component under "Wallet/account-card."

### Design

The components should look like this:

### Edit
I've removed the FlatList as agreed and updated the associated videos accordingly; please review.

##Light Mode

https://github.com/status-im/status-mobile/assets/5999878/7d729c25-b786-4559-9c2c-e1f344ad90e8

##Dark mode

https://github.com/status-im/status-mobile/assets/5999878/36b31ff9-8fca-4d04-8a06-93544e03f386

### CODE REVIEW

File: src/quo2/components/wallet/account_card/style.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/style.cljs]: trailing parentheses

File: src/quo2/components/wallet/account_card/view.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/view.cljs]: trailing parentheses

File: src/quo2/components/wallet/account_card/view.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/view.cljs]: if you use an `if` without a second condition, then you can use `when`

i.e

```
(if condition
     (do-something))
```

should be rewritten as

```
(when condition
   (do-something))
```

we use `if` only if you have two branches:

```
(if condition
   (do-something-1)
   (do-something-2))
```

File: src/quo2/components/wallet/account_card/view.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/view.cljs]: parentheses

File: src/quo2/components/wallet/account_card/view.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/view.cljs]: parentheses

File: src/quo2/core.cljs
Comment (cammellos) [inline src/quo2/core.cljs]: parentheses

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (cammellos) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: p

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (cammellos) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: p

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (cammellos) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: p

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (cammellos) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: p

File: src/quo2/components/wallet/account_card/account_card_spec.cljs
Comment (cammellos) [inline src/quo2/components/wallet/account_card/account_card_spec.cljs]: `defs` generally go top level, so above and outside `(h/describe`
Not sure exactly on this case as `h/describe` is a macro, so just double checking with @ilmotta , but I am fairly certain that you want to have it outside (I'd use `let` if inside, as looks more customary)

Comment (ilmotta) [reply src/quo2/components/wallet/account_card/account_card_spec.cljs]: I agree with @cammellos, the `def` and `defn` should move outside `h/describe` ideally. I understand in Javascript it's customary to define variables and even functions inside each (nested) `describe` block, but in Clojure it doesn't translate as idiomatic code. I also prefer the vars outside to make REPL usage more friendly, since it's a bit easier to eval the top level s-exp.

The local binding I guess doesn't work well here because the value needs to be used by two tests, and surrounding with a `let` both `h/test` makes the code less readable due to nesting (a matter of taste perhaps).

Per @cammellos's suggestion:

```clojure
(def john-doe-name "John Doe")

(defn get-test-data
  [type watch-only]
  {:name                john-doe-name
   :balance             "€1,000.00"
   :percentage-value    "50%"
   :amount              "€500.00"
   :customization-color :blue
   :watch-only          watch-only
   :type                type})

(h/describe "Account_card tests"
  (h/test "Renders normal"
    (let [data (get-test-data :default false)]
      (h/render [account-card/view data])
      (h/is-truthy (h/get-by-text john-doe-name))))

  (h/test "Renders watch-only"
    (let [data (get-test-data :watch-only true)]
      (h/render [account-card/view data])
      (h/is-truthy (h/get-by-text john-doe-name))))

  ...)
```

Also the code is not indented properly and it was confusing for me to read it. @FFFra you can fix this with `make lint-fix`.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (flexsurfer) [inline src/quo2/components/wallet/account_card/view.cljs]: `[:<>`

File: src/quo2/components/wallet/account_card/view.cljs
Comment (flexsurfer) [inline src/quo2/components/wallet/account_card/view.cljs]: :size 20 is default, so its redundant here

File: src/quo2/components/wallet/account_card/view.cljs
Comment (flexsurfer) [inline src/quo2/components/wallet/account_card/view.cljs]: `watch-only?`

File: src/quo2/components/wallet/account_card/view.cljs
Comment (flexsurfer) [inline src/quo2/components/wallet/account_card/view.cljs]: margin should be integer i guess

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/style.cljs]: We follow the Clojure Style Guide, so we have a final `\n` https://guide.clojure.style/#terminate-files-with-a-newline. Please, fix all files in the PR without terminating newline characters.

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: We can use kebab style in this map.

File: src/quo2/components/wallet/account_card/style.cljs
Comment (J-Son89) [inline src/quo2/components/wallet/account_card/style.cljs]: should this be `:army` or customization-color passed as a param?

File: src/quo2/components/wallet/account_card/view.cljs
Comment (J-Son89) [inline src/quo2/components/wallet/account_card/view.cljs:11]: `watch-only` prop is redundant as there is type `watch-only`, you can create an internal let var if you think it make it easier to read.

```
(defn user-account-view
  [{:keys [name balance percentage-value amount customization-color watch-only type]}]
  (let [watch-only? (= :watch-only type)]
  ...
  
 ```

File: src/quo2/components/wallet/account_card/view.cljs
Comment (J-Son89) [inline src/quo2/components/wallet/account_card/view.cljs]: `customization-color` should be a prop that's passed in 👍

File: src/quo2/components/wallet/account_card/view.cljs
Comment (J-Son89) [inline src/quo2/components/wallet/account_card/view.cljs]: `watch-only` prop is redundant as there is type `watch-only`, you can create an internal let var if you think it make it easier to read.

```
(defn user-account-view
  [{:keys [name balance percentage-value amount customization-color type]}]
  (let [watch-only? (= :watch-only type)]
  ...
  
 ```

File: src/quo2/components/wallet/account_card/account_card_spec.cljs
Comment (J-Son89) [inline src/quo2/components/wallet/account_card/account_card_spec.cljs]: namespace should be:
`quo2.components.wallet.account-card.component-spec`

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: Nice catch, tks!

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: tks, its removed

File: src/quo2/components/wallet/account_card/style.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/style.cljs]: Tks, to avoid this in the future I added that to my VSCode Settings

File: src/quo2/components/wallet/account_card/account_card_spec.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/account_card_spec.cljs]: Tks! Already applied

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [reply src/quo2/components/wallet/account_card/style.cljs]: Great, that's the way to go! Thanks

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: In this case I was just following figma

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: Done

**Review by J-Son89 — APPROVED** [review]
nice work @FFFra 👍

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/style.cljs]: If the theme changes the map won't be updated because `colors/theme-colors` relies on a reactive atom.

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: The keyword should use kebab case.

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/style.cljs]: These are not style properties. Please move `size` and `weight` outside this namespace. Vars `metrics`, `account-value` and `add-account-title` also should be changed.

**Review by ilmotta — CHANGES_REQUESTED** [review]

File: src/quo2/components/wallet/account_card/style.cljs
Comment (J-Son89) [reply src/quo2/components/wallet/account_card/style.cljs]: that is can you make this a `defn`. e,g
```
(defn [] add-account-container
  {:width              161
   :height             88
   ...
   ```

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (cammellos) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: just `:d` will do

```
:key-extractor :id
```
since `:id` is already a function as it implements`iFn` https://cljs.github.io/api/cljs.core/IFn 
and I think id is already a string

**Comment by flexsurfer** [comment]
hey @FFFra still working on this ?

**Comment by J-Son89** [comment]
@flexsurfer, yep @FFFra is still working here 👍

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: This was removed due to new implementation. But you are right =)

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: Tks! I will apply it if we decide to also include the flat list to the variants preview

File: src/quo2/components/wallet/account_card/style.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/style.cljs]: Thanks!

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/view.cljs]: This line can be joined with the next one to look like `[user-account-view props]`

**Review by ilmotta — CHANGES_REQUESTED** [review]
@FFFra, the PR looks almost good to me, but there's one important thing I think should be done.

Recently, it was decided that we'll try to get inputs/reviews from designers more often in the development process, even before merging some PRs.

For a designer to review your PR build, it's important you improve the Quo Preview screen to support all the different Figma states. This is also useful for devs to review your component. I tried to review this Account Card component in the UI, but it's way too inconvenient when everything is hardcoded and I have to run your branch locally to make code changes.

Good news is you can find many Quo previews examples in the repo :)

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/view.cljs]: We have some exceptions, but most of the time we try to build Design System components that don't use i18n directly because we want them to be translation agnostic.

In this case, in Figma we see the "Add account" text is in a layer named `Title`, which is a good indicator that when the component is of type `:add-account` it could use an argument named `title`.

Additionally, the second argument to `text` should be in the next line https://guide.clojure.style/#vertically-align-fn-args

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: Dangling parentheses.

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: We should strictly follow the Design System. The Account Card component is a single component with many variations. The fact that it can be displayed in a horizontally scrollable list is not specified in the Figma component, this is actually a feature of the real wallet screen using the component.

This is also important to not confuse designers reviewing the preview area.

Therefore, I think we should remove the flatlist option.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/view.cljs]: Please, make all vars private, except `view`. For `defn` we use `defn-` (not the metadata `^:private`).

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/view.cljs]: Usually we don't append `view` or `component` suffixes unless there's a naming conflict and the function name would shadow a parameter. So just `user-account` is enough.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/view.cljs]: We are gradually making DS components *theme aware* by using `quo2.theme/with-theme`. You can find some examples already, e.g. `quo2.components.dividers.divider-label`.

The way it works is that the HoF passes the current theme set by a React Context, and then in the component you use this `theme` parameter whenever you need to use a color or anything that depends on the current theme. In practice, almost all calls to `colors/theme-colors` should pass a third argument `theme` (the current theme).

So there are some changes you need to make in this file and in the style.cljs one.

**Review by ilmotta — CHANGES_REQUESTED** [review]
@FFFra, thanks for improving the preview area. In this review I noticed some other issues, but hopefully all changes should be minor.

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: I understand your point, but that was required before. I will talk to the team and come up with a decision

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: tks, that should be fixed with lint-fix. Im fixing now.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: I agree, nice catch

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (J-Son89) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: Hmm, there is a bit of confusion here. @ilmotta is right for getting this pr merged.
However @FFFra was asked to add it as part of the interview task - but since the interview task is complete let's go with what we want in the codebase, i.e what @ilmotta is suggesting 👍

Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: To be removed =)

Comment (FFFra) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: It's removed. I also updated the videos for this PR =)

**Comment by FFFra** [comment]
Most of the changes are done, some questions I will clarify asap

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: I see guys, thanks. So next time it would help if we have the requirements up-to-date in the PR description, in case they override the guidelines/conventions.

File: src/status_im2/contexts/quo_preview/wallet/account_card.cljs
Comment (ilmotta) [reply src/status_im2/contexts/quo_preview/wallet/account_card.cljs]: @FFFra, unfortunately for us, we have configured `zprint` in such a way as to respect newlines, which means the tool thinks you want the parentheses in a separate line. The tradeoff is that we give devs more room to format their code, but we often have to catch these formatting issues manually in PRs. Unfortunately `lint-fix` won't help.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (ilmotta) [reply src/quo2/components/wallet/account_card/view.cljs]: Just to make things clearer @FFFra, this is more like a nice to have IMO, not so much of a strong requirement. As I said, we have exceptions in the component lib, so it's fine to leave the code as is.

File: src/quo2/components/wallet/account_card/view.cljs
Comment (FFFra) [reply src/quo2/components/wallet/account_card/view.cljs]: all done =)

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/style.cljs]: This also needs to be fixed just like you did in `text-color`.

File: src/quo2/components/wallet/account_card/style.cljs
Comment (ilmotta) [inline src/quo2/components/wallet/account_card/style.cljs]: This call to `colors/theme-colors` also need to be fixed to accept a `theme` argument.

**Review by ilmotta — APPROVED** [review]
@FFFra, I made just two minor comments that need a fix, but I'm approving now, since the code LGTM.

Before proceeding with the merge, it's important to get a designer to review & approve your PR, particularly because we're introducing a new component in the code. @Francesca-G is the right person for that :)

**Comment by ilmotta** [comment]
@Francesca-G, @FFFra, the PR builds are available now for design review. @cammellos added @FFFra as a contributor in read-only mode, so the builds could be generated.

@FFFra, the lint check failed in the CI, you can fix with `make lint-fix`.

```
Formatting required in file src/quo2/components/wallet/account_card/view.cljs
```

**Review by Francesca-G — COMMENTED** [review]
Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=wkDrBUPxHCuvjPAe-1) with the design review comments @ilmotta @FFFra :)

**Comment by FFFra** [comment]
> Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=wkDrBUPxHCuvjPAe-1) with the design review comments @ilmotta @FFFra :)

Thank you, Francesca. I'm afraid I cannot access the file, could you please grant me access?

Cheers!

**Comment by flexsurfer** [comment]
cc @Francesca-G

**Comment by Francesca-G** [comment]
> > Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=wkDrBUPxHCuvjPAe-1) with the design review comments @ilmotta @FFFra :)
> 
> Thank you, Francesca. I'm afraid I cannot access the file, could you please grant me access?
> 
> Cheers!

Hi @FFFra, sorry for the late response.
Please always mention me when you need a review or you have any question for me!

I spoke to Pedro and he gave your email access to our Figma, so you should be able to view the file now, if not please do let me know :)

**Comment by FFFra** [comment]
> > > Here's the [Figma frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3446%3A227419&mode=design&t=wkDrBUPxHCuvjPAe-1) with the design review comments @ilmotta @FFFra :)
> > 
> > 
> > Thank you, Francesca. I'm afraid I cannot access the file, could you please grant me access?
> > Cheers!
> 
> Hi @FFFra, sorry for the late response. Please always mention me when you need a review or you have any question for me!
> 
> I spoke to Pedro and he gave your email access to our Figma, so you should be able to view the file now, if not please do let me know :)

Thank you! I'm taking and already understood the problem. Gonna add you on Discord just in case I have some questions.

**Comment by flexsurfer** [comment]
hey @FFFra is this still relevant?

**Pull request closed without merging**
